### ISSUE TITLE
Add model.eval() call to CLIP model loader

### ISSUE THREAD

**Issue by mikesvoboda** [mikesvoboda/nemotron-v3-home-security-intelligence#5660]
# Context7 Best Practices Audit Finding

## Current Implementation

CLIP model loaded without setting evaluation mode:

```python
# backend/services/clip_loader.py:48-64
model = CLIPModel.from_pretrained(model_path)
# Missing: model.eval()
```

## Best Practice (Context7)

"Always set model to evaluation mode before inference" - this disables dropout and batch normalization training behavior.

## Gap Analysis

* **Correctness Issue**: Without `model.eval()`:
  * Dropout layers remain active, randomly zeroing outputs
  * Batch normalization uses batch statistics instead of running averages
  * Results are non-deterministic and potentially lower quality
* Florence model correctly calls `self.model.eval()` (ai/florence/model.py:493)
* PersonReID correctly handles this

## Recommended Changes

Add `model.eval()` after loading:

```python
def load_clip_model(model_path: str, device: str = "cuda") -> CLIPModel:
    model = CLIPModel.from_pretrained(model_path)
    model.eval()  # Add this line
    model.to(device)
    return model
```

## Files to Modify

* `backend/services/clip_loader.py` - Add model.eval() after loading

## Acceptance Criteria

- [ ] model.eval() called after CLIPModel.from_pretrained()
- [ ] Verify deterministic outputs from CLIP embeddings
- [ ] Add unit test confirming model is in eval mode
